### PR TITLE
fix(DPLAN-2252): sorting is lost after form submit

### PR DESCRIPTION
**Pull request by sakutademos**
### Ticket
[DPLAN-2252](https://demoseurope.youtrack.cloud/issue/DPLAN-2252/Losch-Aktionen-hebt-Sortierung-anhand-der-Filtern-auf-in-A-Tabelle)

**Description:** This PR provides the option to update the hash or leave it unchanged before submitting the form (the issue with the store name was fixed along this PR)

- [ ] Tests updated/created
- [x] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: client/js/lib/statement/AssessmentTable.js [hunk 1]
```diff
@@ -55,70 +55,73 @@ export default function AssessmentTable () {
...
-            abfrageBox = dpconfirm(Translator.trans('check.entries.marked.copy'))
-            if (abfrageBox === true) {
-              document.getElementsByName('r_action')[0].value = 'copy'
-              document.bpform.submit()
-              formsend = true
-            } else {
-              return false
-            }
-            break
+  function handleFormSubmission (task) {
```
Comment (muellerdemos) [inline client/js/lib/statement/AssessmentTable.js:80]: I really like that so much of the logic was moved into its own function which makes the handling here a lot more easy.

**Review by muellerdemos — APPROVED** [review]
Nice

File: client/js/lib/statement/AssessmentTable.js [hunk 2]
```diff
@@ -55,70 +55,73 @@ export default function AssessmentTable () {
...
-          case 'search':
-          case 'filters':
-          case 'viewMode':
-            document.bpform.submit()
-            formsend = true
-            break
+    switch (task) {
+      case 'delete':
+        abfrageBox = dpconfirm(Translator.trans('check.entries.marked.delete'))
+        if (abfrageBox === true) {
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/lib/statement/AssessmentTable.js]: thx, you are right, done here https://github.com/demos-europe/demosplan-core/pull/3712/commits/bea5336fac206a14f89795224aaccaef5267fc1c

File: client/js/components/statement/assessmentTable/DpEditSelectedItemsMenu.vue [hunk 3]
```diff
@@ -129,8 +129,12 @@ export default {
   },
 
   methods: {
+    capitalizeFirstLetter (str) {
```
[the remark this answers is not in the document]
Comment (sakutademos) [reply client/js/components/statement/assessmentTable/DpEditSelectedItemsMenu.vue:132]: I have the same feeling :) I just think which place is best to store such util: store > core > utils, or something more general?

**Review by salisdemos — APPROVED** [review]
looks legit. Thanks

**Pull request merged**